keras_nlp/tokenizers/unicode_character_tokenizer_test.py

```diff
@@ -0,0 +1,360 @@
+            self.assertAllEqual(tokenize_output[i], exp_outputs[i])
+
+    def test_tokenize_scalar(self):
+        input_data = "ninja"
+        tokenizer = UnicodeCharacterTokenizer()
+        call_output = tokenizer(input_data)
+        tokenize_output = tokenizer.tokenize(input_data)
+
+        self.assertAllEqual(call_output, [110, 105, 110, 106, 97])
+        self.assertAllEqual(tokenize_output, [110, 105, 110, 106, 97])
+
+    def test_dense_output(self):
+        input_data = tf.constant(["ninja", "samurai", "▀▁▂▃"])
+        tokenizer = UnicodeCharacterTokenizer(sequence_length=10)
+        call_output = tokenizer(input_data)
+        self.assertIsInstance(call_output, tf.Tensor)
+        self.assertAllEqual(
+            call_output,
+            [
+                [110, 105, 110, 106, 97, 0, 0, 0, 0, 0],
+                [115, 97, 109, 117, 114, 97, 105, 0, 0, 0],
+                [9600, 9601, 9602, 9603, 0, 0, 0, 0, 0, 0],
+            ],
+        )
+
+    def test_detokenize(self):
+        input_data = tf.ragged.constant(
+            [
+                [110, 105, 110, 106, 97],
+                [115, 97, 109, 117, 114, 97, 105],
+                [9600, 9601, 9602, 9603],
+            ]
+        )
+
+        tokenizer = UnicodeCharacterTokenizer()
+        detokenize_output = tokenizer.detokenize(input_data)
+        self.assertAllEqual(
+            detokenize_output,
+            [
+                b"ninja",
+                b"samurai",
+                b"\xe2\x96\x80\xe2\x96\x81\xe2\x96\x82\xe2\x96\x83",
+            ],
+        )
+
+    def test_detokenize_replace_error(self):
+        # 10000000 is an invalid value
+        input_data = tf.ragged.constant([[110, 105, 10000000, 110, 106, 97]])
+        tokenizer = UnicodeCharacterTokenizer(
+            errors="replace", replacement_char=75
+        )
+        detokenize_output = tokenizer.detokenize(input_data)
+        self.assertAllEqual(detokenize_output, [b"niKnja"])
+
+    def test_detokenize_ignore_error(self):
+        input_data = tf.ragged.constant([[110, 105, 10000000, 110, 106, 97]])
+        tokenizer = UnicodeCharacterTokenizer(errors="ignore")
+        detokenize_output = tokenizer.detokenize(input_data)
+        self.assertAllEqual(detokenize_output, [b"ninja"])
+
+    def test_detokenize_strict_error(self):
+        input_data = tf.ragged.constant([[110, 105, 10000000, 110, 106, 97]])
+        tokenizer = UnicodeCharacterTokenizer(errors="strict")
+        with self.assertRaises(tf.errors.InvalidArgumentError):
+            _ = tokenizer.detokenize(input_data)
+
+    def test_normalization_without_UTF8_valueerror(self):
+        with self.assertRaises(ValueError):
+            _ = UnicodeCharacterTokenizer(
+                errors="strict",
+                input_encoding="UTF-16",
+                normalization_form="NFC",
+            )
+
+    def test_lowercase(self):
+        input_data = tf.constant(["NiNJaS aNd sAmUraI"])
```

Comment:
Can we still try making this example input a little shorter, so black doesn't format the output into 20+ lines?

Reply:
Sure that seems easier to analyze as well!